cordformation/src/main/kotlin/net/corda/plugins/Baseform.kt

```diff
@@ -136,6 +157,45 @@ open class Baseform : DefaultTask() {
         project.delete(directory)
     }
 
+    protected fun generateExcludedWhitelist() {
+        if (excludeWhitelist.isNotEmpty()) {
+            val rootDir = Paths.get(project.projectDir.toPath().resolve(directory).resolve("exclude_whitelist.txt").toAbsolutePath().normalize().toString())
+            Files.write(rootDir, excludeWhitelist)
+         }
+    }
+
+    /** invokes ANT tasks GenKey to generate keyStore (optionally) and SignJar task to sign the generated Cordapp/optionally other Cordapps deployed to nodes */
+    protected fun generateKeystoreAndSignCordappJar() {
+        if (!signing.enabled)
+            return
+
+        val baseKeystoreDir = project.projectDir.toPath().resolve(directory)
+
+        if (signing.generateKeystore) {
+            val genKeyTaskOptions = signing.genKeyTaskOptions(baseKeystoreDir)
+            Files.deleteIfExists(Paths.get(genKeyTaskOptions["keystore"]))
+
+            if (signing.hasDefaultKeystoreOptions(baseKeystoreDir)) {
+                project.logger.warn("Generating default keystore to sign Cordapps '${genKeyTaskOptions["keystore"]}'. " +
+                        "To use custom keystore provide keystore details, see documentation at https://docs.corda.net/generating-a-node.html.")
+            } else {
+                project.logger.warn("Generating keystore to sign Cordapps '${genKeyTaskOptions["keystore"]}'")
```

Comment:
`else` is for `if (signing.hasDefaultKeystoreOptions(baseKeystoreDir))`, the intent is to log when a default keystone is produced (and after addressing the comment print default values),  in case the user configured the task only general shorter info.